### ISSUE TITLE
cute_tiled.h:  warning: ?: using integer constants in boolean context, the expression will always evaluate to ‘true’

### ISSUE THREAD

**Issue by aganm** [RandyGaul/cute_headers#178]
gcc gives me this warning. I just wanted to make sure this wasn't a bug so I post it here.

```
In file included from /home/Projects/test/src/lib/ag/cute_impl.c:2:
/home/Projects/test/src/lib/ag/cute_tiled.h: In function ‘cute_tiled_read_vertex_array_internal’:
/home/Projects/test/src/lib/ag/cute_tiled.h:1530:35: warning: ?: using integer constants in boolean context, the expression will always evaluate to ‘true’ [-Wint-in-bool-context]
 1530 |   cute_tiled_expect(m, swap ? 'y' : 'x');
      |                                   ^
/home/Projects/test/src/lib/ag/cute_tiled.h:1247:43: note: in definition of macro ‘CUTE_TILED_CHECK’
 1247 | #define CUTE_TILED_CHECK(X, Y) do { if (!(X)) { cute_tiled_error_reason = Y; goto cute_tiled_err; } } while (0)
      |                                           ^
/home/Projects/test/src/lib/ag/cute_tiled.h:1530:3: note: in expansion of macro ‘cute_tiled_expect’
 1530 |   cute_tiled_expect(m, swap ? 'y' : 'x');
      |   ^~~~~~~~~~~~~~~~~
```

cute_impl.c
```
#define CUTE_TILED_IMPLEMENTATION                                                               
#include "cute_tiled.h"
```

**Comment by RandyGaul** [comment]
Thank you! I'll look into this. At first glance it seems like maybe the macro is expanding in an unexpected way.

**Comment by chuahou** [comment]
The macro for that line expands to (based on `gcc -E`)

    do { do { if (!(cute_tiled_next(m) == swap ? 'y' : 'x')) { cute_tiled_error_reason = "Found unexpected token (is this a valid JSON file?)."; goto cute_tiled_err; } } while (0); } while (0);

or, indented:
```
do
{
	do
	{
		if (!(cute_tiled_next(m) == swap ? 'y' : 'x'))
		{
			cute_tiled_error_reason = "Found unexpected token (is this a valid JSON file?).";
			goto cute_tiled_err;
		}
	} while (0);
} while (0);
```

This appears to be an issue where `cute_tiled_next(m)` is compared to `swap` and used as the condition for the ternary operator instead of being compared to `swap ? 'y' : 'x'`.

I've added a pull request for my attempt at a fix at #187.

**Comment by aganm** [comment]
Thanks
